### ISSUE TITLE
Already used animated stickers aren't fully animated on sticker selection widget

### ISSUE THREAD

**Issue by robmv** [signalapp/Signal-Android#11022]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When opening the sticker selection widget,  the first tab show the stickers previously used, If you have used animated stickers, only a few frames of these animations are show, less than a half a second. On the tab that shows the sticker packs, all not previously used stickers are fully animated, and again the used ones are partially animated and the are stuck as static. Long pressing over them shows the entire animation preview

### Steps to reproduce
- Send an animated sticker
- Open another conversation
- Open the sticker selection widget
- Go to the sticker pack tab of the previously sent sticker
- Go back to the previously used sticker tab.

**Actual result:** The first time the previously used stickers tab is shown the sticker is animated, when you go to the sticker pack tab, the animation of the used stickers are shown for a a small time and stops. Going back to the previously used stickers, the animation is partial too. Closing the sticker selection widget and reopening doesn't solve the issue, after reopen the used stickers play and stop always. If the conversation is closed and reopened, or open a new one, the bug restarts now the first time the previously used stickers are shown on the selection widget are fully animated.

Note: if the selection widget is open and stickers are stuck, locking with the power button and unlocking, bring the used sticker to life but skipping frames and sometimes accelerated. If animations are ok on the selection widget (immediately after opening it) , locking and unlocking doesn't break anything.

**Expected result:** Used animated stickers should be shown fully animated like all other stickers without problems.

### Device info
**Device:** Pixel 4a (4G)
**Android version:** 11 (Android security update: February 5, 2021)
**Signal version:** 5.4.7

### Link to debug log

https://debuglogs.org/bf1b40b784085cf093344d2bf8b30ea2a34d921eb183539ddf9d23e986a6a797

**Comment by robmv** [comment]
Added a video showing the issue

**Comment by robmv** [comment]
Yes, still relevant, Every update to the sticker selection UI bring their own animation bug. with the current release the problems is:

- Open a new conversation.
- Press the sticker button.
- Browse your list of animated and non animated stickers with everything ok.
- Press the recent sticker (magnifying glass icon).
- Recent stickers that are animated are shown ok.
- Press the back button on the search field or the Android back gesture.
- On the bottom sticker list, animations aren't running until you scroll down and up again.

**Comment by robmv** [comment]
Yes @stale it is still relevant, please call a human to check on it.

**Comment by robmv** [comment]
This is still relevant, a human need to review this, setting a bot to close everything because no one is checking is not very effective to solve bugs.

**Comment by cody-signal** [comment]
Sorry about stalebot, we're aware of this issue, but sadly there isn't a clear/direct fix for it. It has to do with the APNG implementation we are using, but it's pretty much the only one we can use.

**Comment by robmv** [comment]
@cody-signal Look like a internal copy of APNG4Android. Am I right? haver you any details about the problem I can try to reproduce outside of the Signal codebase? or is there any kind of guide for testing a custom build of Signal without breaking the Signal rule of using non approved clients? Are there some kind of testing stage servers for contributors?

**Comment by greyson-signal** [comment]
I don't have any specific insights at the moment. If you want to poke around, feel free to do so with the standard production servers.
